### ISSUE TITLE
Warning XSS vulnerabilty

### ISSUE THREAD

**Issue by bluesurfer** [wenzhixin/bootstrap-table#1854]
Hi

This library seems to be still vulnerable to XSS. 
I am using PHP to populate a table using `json_encode` function.
```javascript
var data = <? echo json_encode(get_some_data_from_db()) ?>
```
Suppose that a user inserts in the database `<script>alert('hello')</script>` then `data` in the HTML code would be something like this:
```javascript
var data = [{"name":"<script>alert(\"hello\")<\/script>"}, {"name": "John"}];
```
And this is fine. It is correctly escaped and not interpreted by javascript. However, when I load data in the table using this library such code is interpreted and ends up sending the injected alert message.

The docs says that there is an `escape`  option but that doesn't seems to make anything to prevent this.

**Comment by thornomad** [comment]
Here is a fiddle that demonstrates the concept (I think):

http://jsfiddle.net/thornomad/e3nk137y/4034/

**Comment by bluesurfer** [comment]
Exactly. Thank you for the fiddle. Hope to find a fix for this.

**Comment by dabros** [comment]
@bluesurfer - interesting note i guess, but misleading title perhaps

The issue here really issnt the table, its the incoming data - NOT user input

If you properly sanitize user input this should never happen, so im having trouble seeing how this is really a problem in that ofcourse any unescaped xss in your DB is a serious problem, but this table merely displays info so cant possibly cause any further issue in the DB.

Basically, if you want to research and suggest simple way to stop this then go ahead, but its not an xss vunerability in bootstrap-table if its coming from the DB, though it would be if user input was being SENT to the DB from the bootstrap-table and not escaping properly.

**Comment by wenzhixin** [comment]
Fixed in #1049
